integration_tests/test_search.py

```diff
@@ -1,25 +1,2078 @@
+        )
+
+    def test_beauty_hacks(self):
+        # NOTE: fails on both versions.  The user was obviously looking for a specific
+        # type of book; ideally, the search results would return at least one relevant
+        # one.  Instead, all of the top results are either books about computer hacking
+        # or romance novels.
+        self.search("beauty hacks",
+        AtLeastOne(subject=re.compile("(self-help|style|grooming|personal)")))
+
+    def test_anxiety(self):
+        self.search(
+            "anxiety",
+            Common(
+                genre=re.compile("(psychology|self-help)"),
+                first_must_match=False
+            )
+        )
+
+    def test_mystery(self):
+        self.search(
+            "mystery",
+            Common(genre="Mystery")
+        )
+
+    def test_agatha_christie_mystery(self):
+        # Genre and author
+        self.search(
+            "agatha christie mystery",
+            [ SpecificGenre(genre="Mystery", author="Agatha Christie"),
+              Common(author="Agatha Christie", threshold=1) ]
+        )
+
+    def test_british_mystery(self):
+        # Genre and keyword
+        self.search(
+            "British mysteries",
+            Common(genre="Mystery", summary=re.compile("british"))
+        )
+
+    def test_music_theory(self):
+        # Keywords
+        self.search(
+            "music theory", Common(
+                genre="Music",
+                subject=re.compile("(music theory|musical theory)")
+            )
+        )
+
+    def test_astrophysics(self):
+        # Keyword
+        self.search(
+            "Astrophysics",
+            Common(
+                genre="Science",
+                subject=re.compile("(astrophysics|astronomy|physics|space|science)")
+            )
+        )
+
+    def test_finance(self):
+        # Keyword
+        self.search(
+            "Finance",
+            Common(
+                genre=re.compile("(business|finance)"), first_must_match=False
+            )
+        )
+
+    def test_constitution(self):
+        # Keyword
+        self.search(
+            "Constitution",
+            Common(
+                genre=re.compile("(politic|history)"), first_must_match=False
+            )
+        )
+
+    def test_supervising(self):
+        # Keyword
+        self.search(
+            "supervising", Common(genre="Education")
+        )
+
+    def test_travel_california(self):
+        # NOTE: this doesn't work on either version of ES, but it's closer on ES6.
+
+        # Keyword
+        self.search(
+            "California",
+            Common(genre=re.compile("(travel|guide|fodors)"), first_must_match=False)
+        )
+
+    def test_travel_florida(self):
+        # NOTE: this doesn't work on either version of ES, but it's closer on ES6.
+
+        # Keyword
+        self.search(
+            "Florida",
+            Common(subject=re.compile("(travel|guide|fodors)"), first_must_match=False)
+        )
```

Comment:
This could also be Lauren Groff's Florida

Reply:
That's true. I changed this to a similar test that isn't likely to have a novel with the same name.